extensions/cornerstone/src/components/WindowLevelActionMenu/WindowLevelActionMenu.tsx

```diff
@@ -1,44 +1,84 @@
-import React, { ReactElement, useCallback, useEffect, useState } from 'react';
+import React, { ReactElement, useCallback, useEffect, useState, useRef } from 'react';
 import { useTranslation } from 'react-i18next';
 import classNames from 'classnames';
-import { AllInOneMenu, SwitchButton, useViewportGrid } from '@ohif/ui';
-import { CommandsManager } from '@ohif/core';
-import { utilities } from '@cornerstonejs/tools';
-import { utilities as csUtils } from '@cornerstonejs/core';
-
-const { ViewportColorbar } = utilities.voi.colorbar;
-const { ColorbarRangeTextPosition } = utilities.voi.colorbar.Enums;
-const { colormap: colormapUtils } = csUtils;
+import { AllInOneMenu, ButtonGroup, SwitchButton, useViewportGrid } from '@ohif/ui';
+import { CommandsManager, ServicesManager } from '@ohif/core';
 
 export type WindowLevelPreset = {
   description: string;
   window: string;
   level: string;
 };
 
+export type ColorMapPreset = {
+  ColorSpace;
+  description: string;
+  RGBPoints;
+  Name;
+};
+
 export type WindowLevelActionMenuProps = {
   viewportId: string;
   element: HTMLElement;
   presets: Record<string, Array<WindowLevelPreset>>;
   verticalDirection: AllInOneMenu.VerticalDirection;
   horizontalDirection: AllInOneMenu.HorizontalDirection;
   commandsManager: CommandsManager;
+  serviceManager: ServicesManager;
+  colormaps: Array<ColorMapPreset>;
+  colorbarWidth: string;
+  colorbarContainerPosition: string;
+  colorbarTickPosition: string;
+  colorbarInitialColormap: string;
 };
 
+const buttons = [
+  {
+    children: 'PET',
+    key: 0,
+    style: {
+      minWidth: '50%',
+    },
+  },
+  {
+    children: 'CT',
+    key: 1,
+    style: {
+      minWidth: '50%',
+    },
+  },
+];
```

Comment:
This should be grabbed from the viewports, we can't assume modality is PT and CT, can be MR and PT fusion etc etc